pmm-app/tests/ia/pages/alertsPage.js

```diff
@@ -0,0 +1,78 @@
+const { I } = inject();
+const assert = require('assert');
+
+const alertRow = (alertName) => `//tr[td[text()="${alertName}"]]`;
+
+module.exports = {
+  url: 'graph/integrated-alerting/alerts',
+  columnHeaders: ['Name', 'Severity', 'State', 'Labels', 'Active Since', 'Last Notified', 'Actions'],
+  elements: {
+    noData: '$table-no-data',
+    alertRow: (alertName) => alertRow(alertName),
+    labelsCell: (alertName) => `${alertRow(alertName)}/td[4]//span`,
+    stateCell: (alertName) => `${alertRow(alertName)}/td[3]`,
+    severityCell: (alertName) => `${alertRow(alertName)}/td[2]`,
+    criticalSeverity: '//td[2]/span[text()="Critical"]',
+    highSeverity: '//td[2]/span[text()="High"]',
+    noticeSeverity: '//td[2]/span[text()="Notice"]',
+    warningSeverity: '//td[2]/span[text()="Warning"]',
+    columnHeaderLocator: (columnHeaderText) => `//th[text()="${columnHeaderText}"]`,
+  },
+  buttons: {
+    // silenceActivate returns silence/activate button locator for a given alert name
+    silenceActivate: (alertName) => `${alertRow(alertName)}[1]/td//button[@data-qa="silence-alert-button"]`,
+  },
+  messages: {
+    noAlertsFound: 'No alerts',
+    successfullySilenced: 'Alert silenced',
+    successfullyActivated: 'Alert activated',
+  },
+
+  async silenceAlert(alertName) {
+    const title = await I.grabAttributeFrom(this.buttons.silenceActivate(alertName), 'title');
+
+    if (title === 'Silence') {
+      const bgColorBeforeAction = await I.grabCssPropertyFrom(
+        `${this.elements.alertRow(alertName)}/td`,
+        'background-color',
+      );
+
+      I.click(this.buttons.silenceActivate(alertName));
+      I.verifyPopUpMessage(this.messages.successfullySilenced);
+      I.seeTextEquals('Silenced', this.elements.stateCell(alertName));
+      const bgColorAfterAction = await I.grabCssPropertyFrom(
```

Comment:
just move this into a new action in adminPage class, `veriryColorChange`, or something similar, make it generic to be re-used anywhere we want. As I see you already use it in two functions.

Reply:
Actually, I don't see the reason to put a single assertion into a separate function.